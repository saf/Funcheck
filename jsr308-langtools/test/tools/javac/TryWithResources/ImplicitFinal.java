/*
 * @test  /nodynamiccopyright/
 * @bug 6911256 6964740 6965277
 * @author Maurizio Cimadamore
 * @summary Test that resource variables are implicitly final
 * @compile/fail/ref=ImplicitFinal.out -XDrawDiagnostics ImplicitFinal.java
 */

import java.io.IOException;

class ImplicitFinal implements AutoCloseable {
    public static void main(String... args) {
        try(ImplicitFinal r = new ImplicitFinal()) {
            r = null; //disallowed
        } catch (IOException ioe) { // Not reachable
            throw new AssertionError("Shouldn't reach here", ioe);
        }
    }


     // A close method, but the class is <em>not</em> Closeable or
     // AutoCloseable.

    public void close() throws IOException {
        throw new IOException();
    }
}
