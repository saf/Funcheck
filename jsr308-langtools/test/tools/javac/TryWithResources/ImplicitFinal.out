ImplicitFinal.java:14:13: compiler.err.twr.resource.may.not.be.assigned: r
1 error
